src/libs/Navigation/NavigationRoot.js

```diff
@@ -50,14 +55,24 @@ function parseAndLogRoute(state) {
 
 const NavigationRoot = (props) => {
     useFlipper(navigationRef);
+    const navigationStateRef = useRef(undefined);
+
+    const updateSavedNavigationStateAndLogRoute = (state) => {
+        navigationStateRef.current = state;
+        props.updateCurrentReportId(state);
+        parseAndLogRoute(state);
```

Comment:
✋ Coming from https://github.com/Expensify/App/issues/20825

Salute for all experts for this navigation reboot 🫡

The `state` can be `undefined` and it's crashed the app when we try to use the navigation state in other methods.

Reply:
Hi! This part has `if` for that now https://github.com/Expensify/App/blob/59973575da78597ccac89781b4d7117110b82078/src/libs/Navigation/NavigationRoot.js#L59C1-L66